presto-main/src/main/java/io/prestosql/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -486,7 +494,30 @@ protected Node visitShowCreate(ShowCreate node, Void context)
                 return singleValueQuery("Create Table", formatSql(createTable).trim());
             }
 
-            throw new UnsupportedOperationException("SHOW CREATE only supported for tables and views");
+            if (node.getType() == SCHEMA) {
+                CatalogSchemaName schemaName = createCatalogSchemaName(session, node, Optional.of(node.getName()));
+                accessControl.checkCanShowCreateSchema(session.toSecurityContext(), schemaName);
+
+                if (!metadata.schemaExists(session, schemaName)) {
```

Comment:
I think we should check existence of object before access control

Reply:
See `visitShowTables`. It checks access first. It feels also generally better to check access first, otherwise you can "poke" the system (by trying schema names) and gain information you should not access to. Currently that might to be an issue with schema, but we might tighten visibility in the future.

Reply:
Wait... No, you're right. Show Tables is different. I'll change it.